lib/sync-settings.coffee

```diff
@@ -24,54 +28,21 @@ module.exports =
 
   serialize: ->
 
-  backup: (cb=null) ->
+  backup: ->
     files = {}
-    if atom.config.get('sync-settings.syncSettings')
-      files["settings.json"] = content: JSON.stringify(atom.config.settings, @filterSettings, '\t')
-    if atom.config.get('sync-settings.syncPackages')
-      files["packages.json"] = content: JSON.stringify(@getPackages(), null, '\t')
-    if atom.config.get('sync-settings.syncKeymap')
-      files["keymap.cson"] = content: (@fileContent atom.keymaps.getUserKeymapPath()) ? "# keymap file (not found)"
-    if atom.config.get('sync-settings.syncStyles')
-      files["styles.less"] = content: (@fileContent atom.styles.getUserStyleSheetPath()) ? "// styles file (not found)"
-    if atom.config.get('sync-settings.syncInit')
-      files["init.coffee"] = content: (@fileContent atom.config.configDirPath + "/init.coffee") ? "# initialization file (not found)"
-    if atom.config.get('sync-settings.syncSnippets')
-      files["snippets.cson"] = content: (@fileContent atom.config.configDirPath + "/snippets.cson") ? "# snippets file (not found)"
 
-    for file in atom.config.get('sync-settings.extraFiles') ? []
-      ext = file.slice(file.lastIndexOf(".")).toLowerCase()
-      cmtstart = "#"
-      cmtstart = "//" if ext in [".less", ".scss", ".js"]
-      cmtstart = "/*" if ext in [".css"]
-      cmtend = ""
-      cmtend = "*/" if ext in [".css"]
-      files[file] =
-        content: (@fileContent atom.config.configDirPath + "/#{file}") ? "#{cmtstart} #{file} (not found) #{cmtend}"
-
-    @createClient().gists.edit
-      id: atom.config.get 'sync-settings.gistId'
-      description: "automatic update by http://atom.io/packages/sync-settings"
-      files: files
-    , (err, res) ->
-      if err
-        console.error "error backing up data: "+err.message, err
-        message = JSON.parse(err.message).message
-        message = 'Gist ID Not Found' if message is 'Not Found'
-        atom.notifications.addError "sync-settings: Error backing up your settings. ("+message+")"
-      else
-        atom.notifications.addSuccess "sync-settings: Your settings were successfully backed up. <br/><a href='"+res.html_url+"'>Click here to open your Gist.</a>"
-      cb?(err, res)
+    for own file, sync of SyncManager.get()
+      files[file] = content: sync.reader()
```

Comment:
it would be better for performance if the interface is async, and then wait on all promises to proceed. a nice mixed api would be to support both results - the actual content or promise. I think the standard promise api can be used, by just supplying the current files array.